minio/minio
# Add a check to prevent removing an actively used tier

## Description
This change prevents removal an actively used remote tier

## Motivation and Context
To avoid users from accidentally removing a remote tier that is being referred to by active ILM rules.

## How to test this PR?
1. Add a remote tier
```
➜  mc git:(storage-class)  ./mc admin tier add s3 minio-tier S3TIER-1 --endpoint https://play.minio.io --access-key Q3AM3UQ867SPQQA43P2F --secret-key zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG --bucket kp-sc-add --prefix s3-infreq
 ```
2. Create a bucket and configure ILM rule on it
```
➜  mc git:(storage-class) ./mc mb minio-tier/mybucket
Bucket created successfully `minio-tier/mybucket`.

➜  mc git:(storage-class) ./mc ilm add --transition-date "2021-01-28" --storage-class "S3TIER-1" minio-tier/mybucket 
Lifecycle configuration rule added with ID `c09ko6va7c2027e2mb40` to minio-tier/mybucket.
```
3. Try to remove the remote tier `S3TIER-1`
```
➜  mc git:(storage-class) ✗ ./mc admin tier rm minio-tier S3TIER-1
mc: <ERROR> Unable to remove remote tier. Specified remote tier is actively used by one or more ILM transition rules.
```
You should see the above error message.
## Types of changes

## Checklist:
